### ISSUE TITLE
Does tpot provide a predict_proba?

### ISSUE THREAD

**Issue by ritviksahajpal** [rhiever/tpot#276]
Can tpot provide a `predict_proba` so that I can link it with libraries that can use predict_proba to produce explanations of how the model is working e.g. https://github.com/marcotcr/lime?

E.g.,

```
from tpot import TPOTClassifier
from sklearn.datasets import load_iris
from sklearn.cross_validation import train_test_split
import numpy as np

iris = load_iris()
X_train, X_test, y_train, y_test = train_test_split(iris.data.astype(np.float64),
    iris.target.astype(np.float64), train_size=0.75, test_size=0.25)

tpot = TPOTClassifier(generations=5, population_size=20, verbosity=2)
tpot.fit(X_train, y_train)
# I would like to use something like tpot.predict_proba
print(tpot.score(X_test, y_test))
tpot.export('tpot_iris_pipeline.py')
```

**Comment by rhiever** [comment]
Not currently, but this sounds like something we could add for the `TPOTClassifier` method. Can you confirm that all sklearn classifiers have a `predict_prob()` method? We would need to double-check that.

**Comment by rhiever** [comment]
@sahilshah1194, this might be a good issue to start with. Can you please look through all of the [classifier operators used in TPOT](https://github.com/rhiever/tpot/tree/master/tpot/operators/classifiers) in the sklearn docs to confirm that all of them have a `predict_prob()` method?

**Comment by ritviksahajpal** [comment]
@rhiever , thanks for considering! I could find predict_proba for all the classifiers at https://github.com/rhiever/tpot/tree/master/tpot/operators/classifiers, except passive aggressive. Also, not sure about XG Boost

**Comment by rhiever** [comment]
We might have to drop the passive aggressive classifier for this issue then, which I'm not entirely opposed to. XGBoost seems to have `predict_prob()`.

**Comment by ritviksahajpal** [comment]
That sounds good, thanks for an excellent package!

**Comment by mfeurer** [comment]
Just a hint, we use a softmax function in auto-sklearn to get probability estimates for the classifiers which don't support predict_proba, see [here](https://github.com/automl/auto-sklearn/blob/master/autosklearn/pipeline/implementations/util.py).

**Comment by rhiever** [comment]
Thanks @mfeurer!

**Comment by ecatkins** [comment]
I don't think the [LinearSVC](http://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html) classifier has a predict_proba either

**Comment by sahilshah1194** [comment]
Looking into it!

**Comment by tvashtar** [comment]
Very glad to see this issue! Just to add this has an additional (and I think free) win. It allows users to define a much more broad score function. Sklearn's [make_scorer ](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.make_scorer.html ) takes a need_proba keyword, that only needs for the classifer(or pipeline) to have a predict_proba method. Simply implementing this on the TPOT classifier should open that up for free. But it's worth a test before announcing of course :) 

This would allow people to use things like [Normalized Discount Cumulative Gain](https://en.wikipedia.org/wiki/Discounted_cumulative_gain), which is useful for a range of problems.

**Comment by rhiever** [comment]
This should be in the next release.
